runtime/parachains/src/builder.rs

```diff
@@ -86,24 +104,40 @@ impl<T: paras_inherent::Config> BenchBuilder<T> {
 	/// of the functions in this implementation.
 	pub(crate) fn new() -> Self {
 		BenchBuilder {
-			// Validators should be declared prior to all other setup.
 			validators: None,
-			// Starting block number; we expect it to get incremented on session setup.
 			block_number: Zero::zero(),
-			// Starting session; we expect it to get incremented on session setup.
 			session: SessionIndex::from(0u32),
-			// Session we want the scenario to take place in. We roll to to this session.
 			target_session: 2u32,
-			// Optionally set the max validators per core; otherwise uses the configuration value.
 			max_validators_per_core: None,
-			// Optionally set the max validators; otherwise uses the configuration value.
 			max_validators: None,
-			// Optionally set the number of dispute statements for each candidate,
 			dispute_statements: BTreeMap::new(),
+			dispute_sessions: Default::default(),
+			backed_and_concluding_cores: Default::default(),
+			code_upgrade: None,
 			_phantom: sp_std::marker::PhantomData::<T>,
 		}
 	}
 
+	/// Set `self.dispute_sessions`.
+	pub(crate) fn set_dispute_sessions(mut self, dispute_sessions: Vec<u32>) -> Self {
```

Comment:
nit: `Vec<T>` as well as `[T]` and `&[T]` impl https://doc.rust-lang.org/std/vec/struct.Vec.html#impl-AsRef%3C%5BT%5D%3E , https://doc.rust-lang.org/std/primitive.slice.html#impl-AsRef%3C%5BT%5D%3E https://doc.rust-lang.org/std/primitive.array.html#impl-AsRef%3C%5BT%5D%3E and hence using `impl AsRef<[T]>` is preferable and avoids all the `vec![..]` clutter.

Reply:
I might be missing some basic understanding here, but using `impl AsRef<[T]>` here seems to make things more complicated because we eventually need the builder struct to own the type. If we have some pointer type instead of a Vec then we need to add lifetimes to the builder struct, which seems unnecessarily complicated. To make this function signature more flexible though I changed it to accept a slice in de60558

Reply:
Priority for test code is, that it's low visual pain. If we can avoid `vec!` in a dozend places, it's worth the internal allocation of `as_ref().to_vec()`.